integration-tests/smoke/vrfv2plus_test.go

```diff
@@ -785,7 +787,61 @@ func TestVRFv2PlusMigration(t *testing.T) {
 	)
 	require.NoError(t, err, "error setting up VRF v2_5 env")
 
+	t.Run("Test migration of Subscription Billing subID", func(t *testing.T) {
+		// Migrate subscription from old coordinator to new coordinator, verify if balances
+		// are moved correctly and requests can be made successfully in the subscription in
+		// new coordinator
+		testSubMigration(subIDs, false, vrfv2PlusContracts, t, l, env, vrfv2PlusData, config,
+			linkAddress, mockETHLinkFeedAddress, nodesMap)
+	})
+
+	t.Run("Test migration of direct billing using VRFV2PlusWrapper subID", func(t *testing.T) {
+		// Migrate wrapper subscription from old coordinator to new coordinator, verify if balances
+		// are moved correctly and requests can be made successfully in the subscription in
+		// new coordinator
+		testSubMigration(subIDs, true, vrfv2PlusContracts, t, l, env, vrfv2PlusData, config,
+			linkAddress, mockETHLinkFeedAddress, nodesMap)
+	})
+}
+
+func testSubMigration(subIDs []*big.Int,
```

Comment:
This will be very debatable and contradicts to DRY principle - but I dont like when tests have logic for the **test case.** (i dont mean logic for functionality)

Just test should be easy read and maintain.

So in this case I would not share test steps between normal migration and direct funding migration

Here is an excerpt from some [post](https://dev.to/mbarzeev/why-practicing-dry-in-tests-is-bad-for-you-j7f) with which I completely agree

> There is a clear purpose for your tests and that is to prevent regressions and provide confidence when making future changes. Your tests should not become a burden to maintain or use.
I much prefer simple tests, where everything which is relevant to a test case can be found between its curly brackets, and I really don’t care if that code repeats itself. It reassures me that there is little chance that this test is affected somehow by any side effect I’m not aware of.

Reply:
Removed common function between normal migration and direct funding migration